### DISCUSSION TITLE
Action button to export all rows from a table into a custom string

### DISCUSSION THREAD

**Discussion by magickaito** [Budibase/budibase#5918] [Help] [answered]
![Screenshot 2022-05-17 at 2 00 24 PM](https://user-images.githubusercontent.com/1195573/168739783-80126fb0-f02a-4c03-8827-71fcb74697cf.jpg)

Hi guys I am learning this fantastic framework. 

So I have a table with rows, and I want to add a "Generate Report" button so that
1) I will first loop through all rows in the table, consolidate the data into a long strings.
2) Then I will send a post HTTP request to a server to generate a PDF report. 

How do I start with step 1?

**Comment by melohagan** [comment, marked as answer]
Hey @magickaito 

You can trigger an automation via actions: https://docs.budibase.com/docs/app-action

You could then combine that with a **Query Rows** automation action, which feeds into an **External Data Connector**.

You can use this binding to get the response from the Query Rows:
<img width="184" alt="Screenshot 2022-05-17 at 10 38 07" src="https://user-images.githubusercontent.com/101575380/168780726-45ddfdcf-4b35-4d83-bd85-b26206fb309b.png">

If you need it as a string, then you could try `JSON.stringify($("steps.1.rows"))` in a JavaScript binding. Some other transformations may be needed for your post request.

More on REST Bindings here: https://docs.budibase.com/docs/rest-bindings

**Note** for internal tables, there is a query limit of 1000 rows.
